### ISSUE TITLE
Feature request: _cat/component_templates endpoint

### ISSUE THREAD

**Issue by ceeeekay** [elastic/elasticsearch#68941]
Hi there,

It would be nice to have a `_cat` endpoint to list component templates in the same way as `_cat/templates`, e.g., `_cat/component_templates`.

While `_cat/templates` lists `composed_of` for any components that are in use by v2 templates, there's no direct way to get a brief list of all template components, including unused components.

`_component_template` will list them but is too verbose when there are a lot of components. This output can be piped through a command line tool such as `jq` for a summary, but this is not possible in the Kibana Dev Tools interface. 

Having to do this from the command line also means dealing with appropriate auth to make the request, whereas Kibana Dev Tools interface can be used by any authorised user.

tl;dr: Getting an answer to the question "What component templates do I have?" seems more complicated than necessary.

Thanks!

**Comment by danhermann** [comment]
Closed by https://github.com/elastic/elasticsearch/pull/71274
